### PR TITLE
GH-41 Add gh action to maven deploy packages

### PR DESCRIPTION
**Pull request by mike-vasylchenko**
Closes https://github.com/softcannery/camunda-formio-accelerator/issues/41

### CODE REVIEW

**Comment by volkotyk** [comment]
looks good =)

**Review by volkotyk — APPROVED** [review]

File: .github/workflows/build-test-and-publish.yaml [hunk 1]
```diff
@@ -69,6 +69,10 @@ jobs:
           NEW_VERSION: ${{ steps.version.outputs.version }}
       - name: test and build JAR
         run: mvn package -ntp
+      - name: Upload new versions in maven registry
+        run: | 
+          cp settings.xml ~/.m2/
```
Comment (softcannery-ltd) [inline .github/workflows/build-test-and-publish.yaml:74]: Let's move settings.xml to .m2 folder in the repository to make it clean and use `--settings` option for `mvn` command to set the alternate path for the user settings file in the workflow

**Review by softcannery-ltd — CHANGES_REQUESTED** [review]

**Comment by softcannery-ltd** [comment]
@mike-vasylchenko It looks the deploy was skipped for deploy: `Skipping artifact deployment`

```
[INFO] 
[INFO] --- maven-deploy-plugin:3.1.1:deploy (default-deploy) @ cucumber-tests ---
[INFO] Skipping artifact deployment
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] camunda-formio-accelerator 0.0.16-SNAPSHOT ......... SUCCESS [  6.491 s]
[INFO] camunda-formio-plugin 0.0.16-SNAPSHOT .............. SUCCESS [  9.268 s]
[INFO] camunda-formio-content 0.0.16-SNAPSHOT ............. SUCCESS [  9.582 s]
[INFO] camunda-formio-example 0.0.16-SNAPSHOT ............. SUCCESS [ 13.572 s]
[INFO] Serenity Cucumber Tests 0.0.1-SNAPSHOT ............. SUCCESS [  6.443 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  45.707 s
[INFO] Finished at: 2023-09-14T23:45:39Z
[INFO] ------------------------------------------------------------------------
```

File: .github/workflows/build-test-and-publish.yaml (hunk 1 above)
Comment (softcannery-ltd) [reply .github/workflows/build-test-and-publish.yaml:74]: Let's also use `-ntp` option to mute download progress.

**Comment by softcannery-ltd** [comment]
> @mike-vasylchenko It looks the deploy was skipped by maven-deploy-plugin: `Skipping artifact deployment`
> 
> ```
> [INFO] 
> [INFO] --- maven-deploy-plugin:3.1.1:deploy (default-deploy) @ cucumber-tests ---
> [INFO] Skipping artifact deployment
> [INFO] ------------------------------------------------------------------------
> [INFO] Reactor Summary:
> [INFO] 
> [INFO] camunda-formio-accelerator 0.0.16-SNAPSHOT ......... SUCCESS [  6.491 s]
> [INFO] camunda-formio-plugin 0.0.16-SNAPSHOT .............. SUCCESS [  9.268 s]
> [INFO] camunda-formio-content 0.0.16-SNAPSHOT ............. SUCCESS [  9.582 s]
> [INFO] camunda-formio-example 0.0.16-SNAPSHOT ............. SUCCESS [ 13.572 s]
> [INFO] Serenity Cucumber Tests 0.0.1-SNAPSHOT ............. SUCCESS [  6.443 s]
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD SUCCESS
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time:  45.707 s
> [INFO] Finished at: 2023-09-14T23:45:39Z
> [INFO] ------------------------------------------------------------------------
> ```

I see it is skipping cucumber-tests. It is OK

**Review by softcannery-ltd — APPROVED** [review]

**Pull request merged**